### DISCUSSION TITLE
How to split a pdf into images

### DISCUSSION THREAD

**Discussion by Tejareddy94** [py-pdf/pypdf#2519] [Q&A] [answered]
How to split a Digital pdf or Scanned pdf into images using pypdf
My use-case is i want to split a pdf into individual images with Good Quality but with less in File Size 
Can any one help with this one?

**Comment by pubpub-zz** [comment, marked as answer]
The manual should help:
https://pypdf.readthedocs.io/en/stable/user/extract-images.html

**Reply by Tejareddy94** [reply]
This will extract images from a page. But what i want is to convert the whole page into an image rather than extracting all the  images present in the page

**Reply by stefan6419846** [reply]
This is out of scope for *pypdf*. Consider looking into the `pdf2image` Python package or the underlying `pdftocairo` tool.

**Reply by Tejareddy94** [reply]
Okay thank you will try for some other package then.

Tried pdf2image package but it is increasing the size of the image to  2-4mb  but `pypdf` is able split images in less than 1mb. it is only working for scanned pdfs but for digital forms it is extracting all the images present in pdf. I thought there is some work around in pypdf package.

**Reply by stefan6419846** [reply]
> but for digital forms it is extracting all the images present in pdf

This is wrong - you are probably mixing this up with `pdfimages`.

**Comment by pubpub-zz** [comment]
This is not an issue I convert it into a discussion

**Reply by Tejareddy94** [reply]
So is there a way to convert a pdf page to image ?

**Reply by pubpub-zz** [reply]
No: you are talking about scanned page : pypdf can extract the images. Pypdf is not a 'viewer' which is doing page rendering
